spring-cloud-sleuth-core/src/main/java/org/springframework/cloud/sleuth/instrument/reactor/ScopePassingSpanSubscriber.java

```diff
@@ -38,18 +39,18 @@
 
 	private static final Log log = LogFactory.getLog(ScopePassingSpanSubscriber.class);
 
-	private final Span span;
+	private final TraceContext spanTraceContext;
 	private final Subscriber<? super T> subscriber;
 	private final Context context;
-	private final Tracer tracer;
+	private final Tracing tracing;
 	private Subscription s;
 
 	ScopePassingSpanSubscriber(Subscriber<? super T> subscriber, Context ctx, Tracing tracing) {
 		this.subscriber = subscriber;
-		this.tracer = tracing.tracer();
+		this.tracing = tracing;
 		Span root = ctx != null ?
-				ctx.getOrDefault(Span.class, this.tracer.currentSpan()) : null;
-		this.span = root;
+				ctx.getOrDefault(Span.class, tracing.tracer().currentSpan()) : null;
```

Comment:
ahh Span is used here.. maybe you'll need the tracer afterall...

Reply:
tracing.tracer().currentSpan() is only called one time by the constructor, so I think there is no need to use tracer